### DISCUSSION TITLE
Replacing use of "[" and "]" as it violates RFC 7230 and RFC 3986

### DISCUSSION THREAD

**Discussion by ManuConstantin** [spatie/laravel-query-builder#852] [Ideas]
Since characters like "[", "]", "{", "}" are not compliant with RFC 7230 and RFC 3986  it would be nice if we could replace them with something else.
Example 
From filter[something] 
To filter%something%

**Comment by Lumethys** [comment]
to my knowledge, this package is built according to the [Json:api specification](jsonapi.org), in which the defined filter syntax, which is defined in their [appendix query details](https://jsonapi.org/format/#appendix-query-details-square-brackets) and [quer parameter family](https://jsonapi.org/format/#query-parameters-families), use the square brackets
